sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneComparer.cs

```diff
@@ -0,0 +1,151 @@
+                {
+                    return false;
+                }
+
+                if (!trackOneEvent.SystemProperties.OrderBy(kvp => kvp.Key).SequenceEqual(trackTwoEvent.SystemProperties.OrderBy(kvp => kvp.Key)))
+                {
+                    return false;
+                }
+            }
+
+            // Since we know that the event bodies and system properties are equal, if the property sets are the
+            // same instance, then we know that the events are equal.  This should only happen if both are null.
+
+            if (Object.ReferenceEquals(trackOneEvent.Properties, trackTwoEvent.Properties))
+            {
+                return true;
+            }
+
+            // If either property is null, then the events are not equal, since we know that they are
+            // not both null.
+
+            if ((trackOneEvent.Properties == null) || (trackTwoEvent.Properties == null))
+            {
+                return false;
+            }
+
+            // The only meaningful comparison left is to ensure that the property sets are equivilent,
+            // the outcome of this check is the final word on equality.
+
+            if (trackOneEvent.Properties.Count != trackTwoEvent.Properties.Count)
+            {
+                return false;
+            }
+
+            return trackOneEvent.Properties.OrderBy(kvp => kvp.Key).SequenceEqual(trackTwoEvent.Properties.OrderBy(kvp => kvp.Key));
+        }
+
+        /// <summary>
+        ///   Compares event position between its representations in track one and
+        ///   track two to determine if the instances represent the same event.
+        /// </summary>
+        ///
+        /// <param name="trackOnePosition">The track one event position to consider.</param>
+        /// <param name="trackTwoPosition">The track two event position to consider.</param>
+        ///
+        /// <returns><c>true</c>, if the two events are structurally equivilent; otherwise, <c>false</c>.</returns>
+        ///
+        public static bool IsEventPositionEquivalent(TrackOne.EventPosition trackOnePosition,
+                                                    EventPosition trackTwoPosition)
+        {
+            // If the positions are the same instance, they're equal.  This should only happen
+            // if both are null, since the types differ.
+
+            if (Object.ReferenceEquals(trackOnePosition, trackTwoPosition))
+            {
+                return true;
+            }
+
+            // If one or the other is null, then they cannot be equal, since we know that
+            // they are not both null.
+
+            if ((trackOnePosition == null) || (trackTwoPosition == null))
+            {
+                return false;
+            }
+
+            // Compare the properties and ensure that they're equal.
+
+            return
+            (
+                (String.Equals(trackOnePosition.Offset, trackTwoPosition.Offset, StringComparison.OrdinalIgnoreCase))
```

Comment:
Why OrdinalIgnoreCase rather than Ordinal? My first thought would be to use Ordinal because these are internally set system properties.

Reply:
Fair point, and they're intended to otherwise be numeric.   I'll swap that over.